queue.go

```diff
@@ -103,3 +103,44 @@ func (w *WaitQueue) Unlock() {
 		w.work.Unlock()
 	}
 }
+
+// BoundedMutuallyExclusiveWaitQueue is a work queue that allows a bounded number of tasks to execute
+// at any one time, but only one of a particular task type
+type BoundedMutuallyExclusiveWaitQueue struct {
+	sem    chan int
+	queues map[string]*WaitQueue
+}
+
+// New returns a new BoundedMutuallyExclusiveWaitQueue ready for use.
+func NewBounded(size int, queueName string, additionalQueueNames ...string) *BoundedMutuallyExclusiveWaitQueue {
```

Comment:
There's no need to define both `queueName` and `additionalQueueNames` here. You can simply define a variadic parameter and check that the length of the resulting array > 0.